pybamm/expression_tree/operations/evaluate.py

```diff
@@ -17,6 +17,57 @@
 
     config.update("jax_enable_x64", True)
 
+    class JaxCooMatrix:
```

Comment:
should this class have a docstring (and corresponding rst file in the docs) or is it deep enough to never be user-facing?

Reply:
users should never have to use this. The only way they would actually get a JaxCooMatrix is if they define an expression to return a sparse matrix, which I don't think we need to consider?

Reply:
however, I really should add docstrings to this! which is now done